### DISCUSSION TITLE
2FA Setup fails .. receiving wrong code in TOTP

### DISCUSSION THREAD

**Discussion by doooooo** [community/community#93931] [Other Feature Feedback, Questions, & Ideas]
Discussion has been moved here:
https://github.com/orgs/community/discussions/93947

I'm using Microsoft authenticator to enable 2FA .. i successfully scanned the QR Code, however the 6 digit code always fails to complete setup.
i'm not able to authenticate thru SMS as well, since my country Egypt is not in the countries list !
please advise !

**Comment by ghostinhershell** [comment]
Hi @doooooo,

Thanks so much for reaching out in the Community! At the moment Egypt isn't a supported country for SMS authentication but we do update this list periodically.

You mentioned that your TOTP app isn't reliably sending you codes that allow you to login to GitHub. I recommend checking the accuracy of the system clock against a known time standard. TOTP stands for Time-based One-Time Password, and requires access to an accurate clock in order for the codes to be generated correctly. In other words, your system time must be accurate and match the timezone that's set in your TOTP app to work.

**Reply by doooooo** [reply]
@ghostinhershell .. Thanks for your reply 
I set my clock accurately to EST time (GMT+2), by enabling Automatic Date and Time on mobile settings.
However, i still get error codes
Is there specific Timezone i should set my mobile to ?
